dashboard/src/app/workspaces/workspace-details/environments/list-agents/list-agents.controller.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2015-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ */
+'use strict';
+
+/**
+ * @ngdoc controller
+ * @name workspace.details.controller:ListAgentsController
+ * @description This class is handling the controller for list of agents
+ * @author Ilya Buziuk
+ */
+export class ListAgentsController {
+
+  /**
+   * Default constructor that is using resource
+   * @ngInject for Dependency injection
+   */
+  constructor(cheAPI) {
+    this.cheAgent = cheAPI.getAgent();
+    
+    this.cheAgent.fetchAgents().then(() => {
```

Comment:
For now I would rather leave it as is but I would probably update it when Agent API would be changed (details in the https://github.com/eclipse/che/issues/1774 discussion)